### ISSUE TITLE
Writing 480 Points in batching writes takes 2 minutes

### ISSUE THREAD

**Issue by Aeroking21** [influxdata/influxdb-client-python#487]
__Steps to reproduce:__

I basically created code that generates points based on a few fields and tags and I simulate the timestamps by adding the seconds feature parameter to `datetime.delta(seconds=)`. These 480 points are then appended to a list and then written in batches of 1000 with all other parameters default. However the whole process of writing these 480 points takes **2min and 25seconds**which is extremely slow for quantity of points. I use the `success_callback` to confirm the batching and uploading of data but it seems to linearly increase in time based on the number of points. I'm not quite sure why this is occurring.

```
import random
import copy
import datetime
from influxdb_client import InfluxDBClient, Point, WriteOptions
from influxdb_client.client.exceptions import InfluxDBError
import time


def generate_list_dictionary():

  # Just fields to modify the template
  fields = ["Price", "Volume", "RSI"]
  companies = ["AAPL", "NFLX", "MSFT", "GOOGL"]

  template = {
    "measurement": "macro",

    "tags": {},

    "fields": {}
  }

  list_points = []

  for timestamp in range(120):
    for company_name in companies:
      for value in fields:
        temp = template
        temp["tags"]["stock"] = company_name
        temp["fields"][value] = random.random()
      
      # Generating simulated timestamps increasing with each outermost for loop
      simulated_time = datetime.datetime.utcnow() + datetime.timedelta(seconds=timestamp) 
      temp["time"] = simulated_time

      dictionary_copy = copy.deepcopy(temp)
      point = Point.from_dict(dictionary_copy, WritePrecision.MS) #Converting it to point
      list_points.append(point)

  return list_points


class BatchingCallback(object):
  
    def success(self, conf: (str, str, str), data: str):
        print(f"Written batch: {conf}, data: {data}")

    def error(self, conf: (str, str, str), data: str, exception: InfluxDBError):
        print(f"Cannot write batch: {conf}, data: {data} due: {exception}")

    def retry(self, conf: (str, str, str), data: str, exception: InfluxDBError):
        print(f"Retryable error occurs for batch: {conf}, data: {data} retry: {exception}")



def write_data(url, token, bucket, org):
  with InfluxDBClient(url=url, token=token, org=org) as client:

    callback = BatchingCallback()

    with client.write_api(write_options=WriteOptions(batch_size=1000),
                                       success_callback=callback.success,
                                       error_callback=callback.error,
                                       retry_callback=callback.retry) as write_api:


      data = generate_list_dictionary() # Generating the data

      # Writing the data in batches
      start = time.perf_counter()
      write_api.write(bucket=bucket, org=org, record=data)
      end = time.perf_counter()

      print(f"Writing time is: {end-start}")


if __name__ == '__main__':

  url = ""
  token = ""
  bucket = ""
  org = ""
  
  write_data(url, token, bucket, org)

```

__Expected behavior:__
I would expect the `success_callback` to print out the batches it has written to the database and upload these in a couple of seconds.

__Actual behavior:__
The `success_callback` does print out what I expect but the timing functionality is clearly wrong on batching, when I tested on `SYNCHRONOUS`  with a simple write, it seemed to work fine:
 ![Screen Shot 2022-08-09 at 12 04 15](https://user-images.githubusercontent.com/55886814/183632625-5b93ea81-4ac9-4409-9182-2551f5021a86.png)

But if I wait **30 seconds** and perform a simple flux query in the UI/client there would still be a lot of data that hasn't been uploaded yet. As I mentioned earlier, I tested it out with 1000 points and it took 5min to upload the data which seems to be quite long. 

I'm not quite sure if I'm misunderstanding the batching functionality or there is something else wrong with my code. But 2min seems an awful long amount of time for just 480 points. Any guidance would be greatly appreciated.

__Specifications:__
 - Client Version:  2.0 - 1.3.0 build
 - InfluxDB Version: 2.3.0
 - Platform: Windows

**Comment by bednar** [comment]
Hi @Aeroking21,

thanks for using our client.

Can you update to the latest version of the client? 

> __Actual behavior:__
> The `success_callback` does print out what I expect:
>  ![Screen Shot 2022-08-09 at 12 04 15](https://user-images.githubusercontent.com/55886814/183632625-5b93ea81-4ac9-4409-9182-2551f5021a86.png)

It is strange behaviour because you have only `480` points, so there should be only one batch.

I've used following code for testing:

```python
import random
import copy
import datetime
from influxdb_client import InfluxDBClient, Point, WriteOptions, WritePrecision
from influxdb_client.client.exceptions import InfluxDBError
import time


def generate_list_dictionary():
    # Just fields to modify the template
    fields = ["Price", "Volume", "RSI"]
    companies = ["AAPL", "NFLX", "MSFT", "GOOGL"]

    template = {
        "measurement": "macro",

        "tags": {},

        "fields": {}
    }

    list_points = []

    for timestamp in range(120):
        for company_name in companies:
            for value in fields:
                temp = template
                temp["tags"]["stock"] = company_name
                temp["fields"][value] = random.random()

            # Generating simulated timestamps increasing with each outermost for loop
            simulated_time = datetime.datetime.utcnow() + datetime.timedelta(seconds=timestamp)
            temp["time"] = simulated_time

            dictionary_copy = copy.deepcopy(temp)
            point = Point.from_dict(dictionary_copy, WritePrecision.MS)  # Converting it to point
            list_points.append(point)

    return list_points


class BatchingCallback(object):

    def success(self, conf: (str, str, str), data: str):
        print(f"Written batch: {conf}, data: {data}")

    def error(self, conf: (str, str, str), data: str, exception: InfluxDBError):
        print(f"Cannot write batch: {conf}, data: {data} due: {exception}")

    def retry(self, conf: (str, str, str), data: str, exception: InfluxDBError):
        print(f"Retryable error occurs for batch: {conf}, data: {data} retry: {exception}")


def write_data(url, token, bucket, org):
    with InfluxDBClient(url=url, token=token, org=org) as client:
        callback = BatchingCallback()

        with client.write_api(write_options=WriteOptions(batch_size=1000),
                              success_callback=callback.success,
                              error_callback=callback.error,
                              retry_callback=callback.retry) as write_api:
            data = generate_list_dictionary()  # Generating the data

            # Writing the data in batches
            start = time.perf_counter()
            write_api.write(bucket=bucket, org=org, record=data)
            end = time.perf_counter()

        print(f"Writing time is: {end - start}")
    print("disposed client.")


if __name__ == '__main__':
    url = "http://localhost:8086"
    token = "my-token"
    org = "my-org"
    bucket = "my-bucket"

    write_data(url, token, bucket, org)

```


and everything works correctly:

```
Written batch: ('my-bucket', 'my-org', 'ms'), data: b'macro,stock=AAPL Price=0.042627620887705,RSI=0.2884896677998039,Volume=0.3484976524747144 1660044392846\nmacro,stock=NFLX Price=0.2646091098428943,RSI=0.5457780664487549,Volume=0.6210015650248878 1660044392846\nmacro,stock=MSFT Price=0.5640741483500665,RSI=0.64626691141506,Volume=0.6323720488172728 1660044392846\nmacro,stock=GOOGL Price=0.9828194684295111,RSI=0.7985436103897493,Volume=0.7557491460991819 1660044392846\nmacro,stock=AAPL Price=0.5044917689232785,RSI=0.5779864608286265,Volume=0.06297234636903881 1660044393846\nmacro,stock=NFLX Price=0.820795392396395,RSI=0.1517432121639608,Volume=0.5324816345262026 1660044393846\nmacro,stock=MSFT Price=0.7058035751151468,RSI=0.4353693555242032,Volume=0.4841274415983293 1660044393846\nmacro,stock=GOOGL Price=0.742650144910845,RSI=0.508311934073671,Volume=0.7394961512120314 1660044393846\nmacro,stock=AAPL Price=0.28132204845833675,RSI=0.4608460764665028,Volume=0.01025397703259201 1660044394846\nmacro,stock=NFLX Price=0.9031977553754655,RSI=0.48593661935208354,Volume=0.5275264190737563 1660044394846\nmacro,stock=MSFT Price=0.20943044142746658,RSI=0.9603151005431695,Volume=0.0014001459239121328 1660044394846\nmacro,stock=GOOGL Price=0.049223601389094984,RSI=0.44904530928635833,Volume=0.37521151120882557 1660044394846\nmacro,stock=AAPL Price=0.3311922803604913,RSI=0.3587334147536698,Volume=0.567806987098132 1660044395846\nmacro,stock=NFLX Price=0.37784712957462285,RSI=0.3178556224579071,Volume=0.9908998758561376 1660044395846\nmacro,stock=MSFT Price=0.17843674264962095,RSI=0.6174811747768745,Volume=0.9932413765406513 1660044395846\nmacro,stock=GOOGL Price=0.6327454491863996,RSI=0.6992580058079171,Volume=0.30892090213558887 1660044395846\nmacro,stock=AAPL Price=0.9654570218573785,RSI=0.9952558003102948,Volume=0.06298267215733377 1660044396846\nmacro,stock=NFLX Price=0.3702951071210645,RSI=0.8772956892979284,Volume=0.8596503958800921 1660044396846\nmacro,stock=MSFT Price=0.4472288474598449,RSI=0.5496785112843201,Volume=0.7612888810025206 1660044396846\nmacro,stock=GOOGL Price=0.7225091115444282,RSI=0.683022843465272,Volume=0.3873725334293051 1660044396846\nmacro,stock=AAPL Price=0.5633335400210903,RSI=0.9766406314641796,Volume=0.5468492885438531 1660044397846\nmacro,stock=NFLX Price=0.16231514045061968,RSI=0.4051980953466019,Volume=0.4059108382187743 1660044397846\nmacro,stock=MSFT Price=0.10986732821530254,RSI=0.6236061552128779,Volume=0.8875622873901002 1660044397846\nmacro,stock=GOOGL Price=0.5271598711435529,RSI=0.8279790686241308,Volume=0.3625478509602089 1660044397846\nmacro,stock=AAPL Price=0.05156030427608049,RSI=0.8931958429680066,Volume=0.1314202497212298 1660044398846\nmacro,stock=NFLX Price=0.9946042563333474,RSI=0.7636238710472691,Volume=0.13693776960637816 1660044398846\nmacro,stock=MSFT Price=0.8860630818508952,RSI=0.2787559318247217,Volume=0.9575354401116307 1660044398846\nmacro,stock=GOOGL Price=0.473856823020279,RSI=0.49253319520638794,Volume=0.9865966925540461 1660044398846\nmacro,stock=AAPL Price=0.9416483885378989,RSI=0.25155991063050953,Volume=0.13113537006943832 1660044399846\nmacro,stock=NFLX Price=0.07514119059991609,RSI=0.5848930467523873,Volume=0.2109418220118474 1660044399846\nmacro,stock=MSFT Price=0.966652608507283,RSI=0.3819892030848281,Volume=0.7653215328113324 1660044399846\nmacro,stock=GOOGL Price=0.5207919472971398,RSI=0.2705397421688185,Volume=0.2119240917201496 1660044399846\nmacro,stock=AAPL Price=0.5272227406508762,RSI=0.12445864034899712,Volume=0.7934086410666465 1660044400846\nmacro,stock=NFLX Price=0.4810495384200244,RSI=0.3477395012880903,Volume=0.19578573082626494 1660044400846\nmacro,stock=MSFT Price=0.2212472964346316,RSI=0.9445008608359543,Volume=0.7366174027978841 1660044400846\nmacro,stock=GOOGL Price=0.6953272292364214,RSI=0.18718817951335065,Volume=0.4178196753558219 1660044400846\nmacro,stock=AAPL Price=0.6945130628358813,RSI=0.5085851594410731,Volume=0.6626795914026129 1660044401846\nmacro,stock=NFLX Price=0.6072635547936774,RSI=0.5945842750763728,Volume=0.17771388872647764 1660044401846\nmacro,stock=MSFT Price=0.8295102897292393,RSI=0.42078916326701665,Volume=0.6974982680406133 1660044401846\nmacro,stock=GOOGL Price=0.548957237838837,RSI=0.7399022744168575,Volume=0.40457408154627894 1660044401846\nmacro,stock=AAPL Price=0.43822280541572545,RSI=0.24552481048369013,Volume=0.4950583973770385 1660044402846\nmacro,stock=NFLX Price=0.5468529962801074,RSI=0.45643988955574455,Volume=0.7029486509905818 1660044402846\nmacro,stock=MSFT Price=0.3728637752989047,RSI=0.6089014618285875,Volume=0.7491343923458146 1660044402846\nmacro,stock=GOOGL Price=0.921942129945669,RSI=0.22301639599986134,Volume=0.4856051067406457 1660044402847\nmacro,stock=AAPL Price=0.7218837009861309,RSI=0.06414767133860144,Volume=0.9264901526128818 1660044403847\nmacro,stock=NFLX Price=0.5873117233840551,RSI=0.17114902233739027,Volume=0.012213592949169016 1660044403847\nmacro,stock=MSFT Price=0.3815836167978951,RSI=0.5087834415939774,Volume=0.17483960234111673 1660044403847\nmacro,stock=GOOGL Price=0.33513009906221536,RSI=0.7382774167469079,Volume=0.6235575359237097 1660044403847\nmacro,stock=AAPL Price=0.5204768674605059,RSI=0.8589804482292279,Volume=0.6921670873845839 1660044404847\nmacro,stock=NFLX Price=0.06217648680848398,RSI=0.5431395303603647,Volume=0.1533733763674846 1660044404847\nmacro,stock=MSFT Price=0.8126685868374074,RSI=0.06781592943913028,Volume=0.04212012819583344 1660044404847\nmacro,stock=GOOGL Price=0.1722586546531789,RSI=0.11752305271288765,Volume=0.7078923806570238 1660044404847\nmacro,stock=AAPL Price=0.7501506070016067,RSI=0.9646816186894887,Volume=0.12548850137950096 1660044405847\nmacro,stock=NFLX Price=0.38778433314554894,RSI=0.20704533238026213,Volume=0.2739603173075429 1660044405847\nmacro,stock=MSFT Price=0.380701598144626,RSI=0.4919910477753119,Volume=0.6316902352451581 1660044405847\nmacro,stock=GOOGL Price=0.515377809334825,RSI=0.3149292925821804,Volume=0.7046756851405729 1660044405847\nmacro,stock=AAPL Price=0.5821803476754869,RSI=0.5633974890212883,Volume=0.5132410115357868 1660044406847\nmacro,stock=NFLX Price=0.5481372823524818,RSI=0.26668479962193625,Volume=0.23762731860409614 1660044406847\nmacro,stock=MSFT Price=0.9440931890585289,RSI=0.8348617409007718,Volume=0.31943883812784424 1660044406847\nmacro,stock=GOOGL Price=0.9169389606959839,RSI=0.15502521432860683,Volume=0.9910953511533184 1660044406847\nmacro,stock=AAPL Price=0.8135804808240928,RSI=0.8790388567968515,Volume=0.0643608017785835 1660044407847\nmacro,stock=NFLX Price=0.8872892763172501,RSI=0.4231468762882614,Volume=0.4453027605688016 1660044407847\nmacro,stock=MSFT Price=0.7221371762003327,RSI=0.6301610571833501,Volume=0.8486556758694873 1660044407847\nmacro,stock=GOOGL Price=0.05821363599648699,RSI=0.5117292521122483,Volume=0.04504116923413071 1660044407847\nmacro,stock=AAPL Price=0.875788205772497,RSI=0.7253506224740421,Volume=0.6302448555510028 1660044408847\nmacro,stock=NFLX Price=0.10666174277491214,RSI=0.8578004914585682,Volume=0.018632516680665567 1660044408847\nmacro,stock=MSFT Price=0.9044849576436617,RSI=0.6902565495282632,Volume=0.2419890676524612 1660044408847\nmacro,stock=GOOGL Price=0.03222785041979093,RSI=0.30976865698571365,Volume=0.38800511394316617 1660044408847\nmacro,stock=AAPL Price=0.7722687046164021,RSI=0.6260994867615622,Volume=0.4816935881273883 1660044409847\nmacro,stock=NFLX Price=0.7241231463274048,RSI=0.7671422615529537,Volume=0.01576926940842438 1660044409847\nmacro,stock=MSFT Price=0.3457582753483738,RSI=0.6641355572649871,Volume=0.639177256950503 1660044409847\nmacro,stock=GOOGL Price=0.9921048397695079,RSI=0.5992202065591596,Volume=0.18193449259832029 1660044409847\nmacro,stock=AAPL Price=0.028706258838278043,RSI=0.10291431382468663,Volume=0.7999137615111457 1660044410847\nmacro,stock=NFLX Price=0.46669711535548963,RSI=0.7738223796617527,Volume=0.6939413709372535 1660044410847\nmacro,stock=MSFT Price=0.6167447314671137,RSI=0.840788816869085,Volume=0.30958966129431775 1660044410847\nmacro,stock=GOOGL Price=0.19893218940164903,RSI=0.7688690990086505,Volume=0.9406235019608121 1660044410847\nmacro,stock=AAPL Price=0.8018254231087228,RSI=0.48461553794624956,Volume=0.9643055812504692 1660044411847\nmacro,stock=NFLX Price=0.15818794706147965,RSI=0.6713317047240424,Volume=0.26385282209664895 1660044411847\nmacro,stock=MSFT Price=0.6253423487385145,RSI=0.6344126303728834,Volume=0.22474014159631162 1660044411847\nmacro,stock=GOOGL Price=0.2700359692126878,RSI=0.5202815713979095,Volume=0.23272969459312443 1660044411847\nmacro,stock=AAPL Price=0.5817145810655969,RSI=0.3184070883990321,Volume=0.8454380345477099 1660044412847\nmacro,stock=NFLX Price=0.8029381451831488,RSI=0.3103749381929519,Volume=0.6507750390321323 1660044412847\nmacro,stock=MSFT Price=0.5666299429194095,RSI=0.20653471163286719,Volume=0.24147443981658911 1660044412847\nmacro,stock=GOOGL Price=0.9510418418269923,RSI=0.5747889671395274,Volume=0.6723787289535473 1660044412847\nmacro,stock=AAPL Price=0.7224390643976563,RSI=0.0987441026717748,Volume=0.34166805607197215 1660044413847\nmacro,stock=NFLX Price=0.9370777128093655,RSI=0.20039634264010864,Volume=0.9286679016271654 1660044413847\nmacro,stock=MSFT Price=0.8175464388427132,RSI=0.14271933849996843,Volume=0.9752972294721297 1660044413847\nmacro,stock=GOOGL Price=0.036837286176415796,RSI=0.07944529274938916,Volume=0.639102628882552 1660044413847\nmacro,stock=AAPL Price=0.5302589611876134,RSI=0.7996362487099271,Volume=0.4582828922629718 1660044414847\nmacro,stock=NFLX Price=0.5709622300962347,RSI=0.4790837280181638,Volume=0.006114275723819951 1660044414847\nmacro,stock=MSFT Price=0.7351359899445159,RSI=0.8836331920823003,Volume=0.10224750891142664 1660044414847\nmacro,stock=GOOGL Price=0.5632870798506894,RSI=0.3590376154401118,Volume=0.6549354736701943 1660044414847\nmacro,stock=AAPL Price=0.6603220869657115,RSI=0.8257303265274107,Volume=0.03500862849561204 1660044415847\nmacro,stock=NFLX Price=0.946645208839842,RSI=0.329758586603513,Volume=0.10427621041895108 1660044415847\nmacro,stock=MSFT Price=0.8070519488836468,RSI=0.02077301823700428,Volume=0.992846773287193 1660044415847\nmacro,stock=GOOGL Price=0.5105939660907224,RSI=0.6546980492326463,Volume=0.20257496425701982 1660044415847\nmacro,stock=AAPL Price=0.18292913258843946,RSI=0.8875781733805367,Volume=0.45083457292729756 1660044416847\nmacro,stock=NFLX Price=0.2219088674454388,RSI=0.488356499905065,Volume=0.27795910513849087 1660044416847\nmacro,stock=MSFT Price=0.26721178009987134,RSI=0.26098492429728615,Volume=0.3073270391881445 1660044416848\nmacro,stock=GOOGL Price=0.7163755464975643,RSI=0.31668923231033086,Volume=0.7363637145693184 1660044416848\nmacro,stock=AAPL Price=0.4299545909551037,RSI=0.889786075195423,Volume=0.8055826983489822 1660044417848\nmacro,stock=NFLX Price=0.4977036081839642,RSI=0.77792161149819,Volume=0.7550302744136642 1660044417848\nmacro,stock=MSFT Price=0.5513683340974062,RSI=0.7047795237508073,Volume=0.7856162344727959 1660044417848\nmacro,stock=GOOGL Price=0.2894354239513798,RSI=0.8866674711100149,Volume=0.5783219325556112 1660044417848\nmacro,stock=AAPL Price=0.7217462982708258,RSI=0.1105311648424685,Volume=0.8862888270702963 1660044418848\nmacro,stock=NFLX Price=0.19154181667245018,RSI=0.18592100860148575,Volume=0.4568132754080818 1660044418848\nmacro,stock=MSFT Price=0.7912978492946694,RSI=0.43245627374303275,Volume=0.5704677051151316 1660044418848\nmacro,stock=GOOGL Price=0.8260995320223212,RSI=0.8472444034077717,Volume=0.19835003545161323 1660044418848\nmacro,stock=AAPL Price=0.17822397047602556,RSI=0.2959510240055846,Volume=0.6507406387042735 1660044419848\nmacro,stock=NFLX Price=0.1500794676622378,RSI=0.10547043813536328,Volume=0.10808737057637641 1660044419848\nmacro,stock=MSFT Price=0.039720011998743865,RSI=0.6820125488873914,Volume=0.6082358017932363 1660044419848\nmacro,stock=GOOGL Price=0.05201688809308336,RSI=0.6289974334276456,Volume=0.2774317062755659 1660044419848\nmacro,stock=AAPL Price=0.659338334771174,RSI=0.7913254809426499,Volume=0.1482715259562135 1660044420848\nmacro,stock=NFLX Price=0.03929312294824061,RSI=0.8519405091833421,Volume=0.6447033589019443 1660044420848\nmacro,stock=MSFT Price=0.6940427487350601,RSI=0.1933623158754859,Volume=0.5774662734927102 1660044420848\nmacro,stock=GOOGL Price=0.43664894208235205,RSI=0.7549451722566657,Volume=0.02527015022996293 1660044420848\nmacro,stock=AAPL Price=0.8479054104934264,RSI=0.5504240295244196,Volume=0.9640850286431738 1660044421848\nmacro,stock=NFLX Price=0.9300570398592962,RSI=0.1855177344557436,Volume=0.9523711492842837 1660044421848\nmacro,stock=MSFT Price=0.4996166290039532,RSI=0.5819580336472222,Volume=0.10898308423785819 1660044421848\nmacro,stock=GOOGL Price=0.5742523394516285,RSI=0.3894042051413439,Volume=0.07253670268984969 1660044421848\nmacro,stock=AAPL Price=0.8346551712476306,RSI=0.7493258589086256,Volume=0.8318280552979613 1660044422848\nmacro,stock=NFLX Price=0.11354599810823096,RSI=0.5988115455593069,Volume=0.21421329013175472 1660044422848\nmacro,stock=MSFT Price=0.5634848517842366,RSI=0.4102572737561839,Volume=0.3957540625387632 1660044422848\nmacro,stock=GOOGL Price=0.3129814119198976,RSI=0.5184629180356637,Volume=0.3783065772806272 1660044422848\nmacro,stock=AAPL Price=0.37405440752104946,RSI=0.9597189626786496,Volume=0.9775027788931323 1660044423848\nmacro,stock=NFLX Price=0.8510303893731224,RSI=0.7073858435415783,Volume=0.13502206382726456 1660044423848\nmacro,stock=MSFT Price=0.9107407555331666,RSI=0.8602258301135207,Volume=0.10911097670401326 1660044423848\nmacro,stock=GOOGL Price=0.16174977413678604,RSI=0.930287369424045,Volume=0.2823695561176852 1660044423848\nmacro,stock=AAPL Price=0.9469243144692507,RSI=0.1013609411343247,Volume=0.9348538764853399 1660044424848\nmacro,stock=NFLX Price=0.32872974564680946,RSI=0.8884816009850656,Volume=0.4003878532500551 1660044424848\nmacro,stock=MSFT Price=0.9234605335441708,RSI=0.9425895989642176,Volume=0.9459408657429648 1660044424848\nmacro,stock=GOOGL Price=0.5085474237052767,RSI=0.6937212864845623,Volume=0.9679032737427165 1660044424848\nmacro,stock=AAPL Price=0.11691626284268386,RSI=0.5691695639166257,Volume=0.0387895859971199 1660044425848\nmacro,stock=NFLX Price=0.20153822857644077,RSI=0.14008908540189735,Volume=0.4424690069741798 1660044425848\nmacro,stock=MSFT Price=0.3521769122633026,RSI=0.2819284996412279,Volume=0.9637509818509227 1660044425848\nmacro,stock=GOOGL Price=0.23669433213404956,RSI=0.7338126325051746,Volume=0.6786604777988858 1660044425848\nmacro,stock=AAPL Price=0.43982878190550645,RSI=0.6631596244203006,Volume=0.8715392961951682 1660044426848\nmacro,stock=NFLX Price=0.27648345302399113,RSI=0.9374422724731347,Volume=0.7552777174460316 1660044426848\nmacro,stock=MSFT Price=0.7465115596750304,RSI=0.31925707289808114,Volume=0.8927439020529413 1660044426848\nmacro,stock=GOOGL Price=0.34675000158809255,RSI=0.664181420895612,Volume=0.967413537872921 1660044426848\nmacro,stock=AAPL Price=0.455315534896911,RSI=0.28168221556651984,Volume=0.6607385294685998 1660044427848\nmacro,stock=NFLX Price=0.15977723156550094,RSI=0.3316605302783452,Volume=0.34678805097735654 1660044427848\nmacro,stock=MSFT Price=0.9579235466750518,RSI=0.06293416575708466,Volume=0.2987338866935557 1660044427848\nmacro,stock=GOOGL Price=0.773339121950102,RSI=0.3587585634289795,Volume=0.4146606402920139 1660044427848\nmacro,stock=AAPL Price=0.5448030483249582,RSI=0.414904792215004,Volume=0.7684333566720053 1660044428848\nmacro,stock=NFLX Price=0.715046119650918,RSI=0.6245664977723786,Volume=0.20538786764507422 1660044428848\nmacro,stock=MSFT Price=0.9125202465848551,RSI=0.32770344552181563,Volume=0.4108997785223536 1660044428848\nmacro,stock=GOOGL Price=0.22141138783853676,RSI=0.8960432948278388,Volume=0.4773263724029232 1660044428848\nmacro,stock=AAPL Price=0.25582820714411303,RSI=0.4488428523762673,Volume=0.5063673892386752 1660044429848\nmacro,stock=NFLX Price=0.9007783233960375,RSI=0.36641591472970725,Volume=0.7676666665733018 1660044429848\nmacro,stock=MSFT Price=0.7759023619429399,RSI=0.6746897482148458,Volume=0.27093575281275883 1660044429848\nmacro,stock=GOOGL Price=0.25439282631666427,RSI=0.019927910870437437,Volume=0.08468109010462199 1660044429848\nmacro,stock=AAPL Price=0.3518626855111421,RSI=0.1964186195949933,Volume=0.23492538179252465 1660044430848\nmacro,stock=NFLX Price=0.12059729870655345,RSI=0.5942764509528834,Volume=0.131083916986785 1660044430848\nmacro,stock=MSFT Price=0.8859364511885498,RSI=0.09278823817701765,Volume=0.010125146932335016 1660044430848\nmacro,stock=GOOGL Price=0.6982347108642404,RSI=0.43376590017261396,Volume=0.39779345695901835 1660044430849\nmacro,stock=AAPL Price=0.4165086878360851,RSI=0.6836659121360394,Volume=0.1882785124923425 1660044431849\nmacro,stock=NFLX Price=0.037012383391132,RSI=0.2739217976789906,Volume=0.6199228799333962 1660044431849\nmacro,stock=MSFT Price=0.42588056674153785,RSI=0.01791693959585483,Volume=0.9735759231745241 1660044431849\nmacro,stock=GOOGL Price=0.20595049401319654,RSI=0.3695453906910343,Volume=0.055710404292897975 1660044431849\nmacro,stock=AAPL Price=0.8594027263513592,RSI=0.9828631957568421,Volume=0.009035717154037282 1660044432849\nmacro,stock=NFLX Price=0.49833536620550745,RSI=0.8809117969580744,Volume=0.25906011069605206 1660044432849\nmacro,stock=MSFT Price=0.6945288481712107,RSI=0.624623119281628,Volume=0.36010264155737226 1660044432849\nmacro,stock=GOOGL Price=0.9204475191572099,RSI=0.4339495387388709,Volume=0.565394132497764 1660044432849\nmacro,stock=AAPL Price=0.2463292964513356,RSI=0.8553820860899616,Volume=0.6389234325011488 1660044433849\nmacro,stock=NFLX Price=0.7480286969316531,RSI=0.9056323504967081,Volume=0.2782869828553748 1660044433849\nmacro,stock=MSFT Price=0.8477673771414326,RSI=0.5620487006822786,Volume=0.5664233819417268 1660044433849\nmacro,stock=GOOGL Price=0.5494003275711895,RSI=0.9082046626983213,Volume=0.28623470857394995 1660044433849\nmacro,stock=AAPL Price=0.22695265702302936,RSI=0.6917467707082156,Volume=0.8288634524777547 1660044434849\nmacro,stock=NFLX Price=0.8843486438781857,RSI=0.24562998282354087,Volume=0.6567776492594464 1660044434849\nmacro,stock=MSFT Price=0.8339184324173537,RSI=0.85294730146537,Volume=0.8071357091924835 1660044434849\nmacro,stock=GOOGL Price=0.38394945795640456,RSI=0.03165005776196783,Volume=0.16189956471536637 1660044434849\nmacro,stock=AAPL Price=0.5082059131142131,RSI=0.8793386905791711,Volume=0.507245907227996 1660044435849\nmacro,stock=NFLX Price=0.36393050574174934,RSI=0.33440340356637555,Volume=0.7351603737399409 1660044435849\nmacro,stock=MSFT Price=0.3594627266582435,RSI=0.6864697733289141,Volume=0.4294125660227397 1660044435849\nmacro,stock=GOOGL Price=0.8459510506215759,RSI=0.0013905476887369206,Volume=0.8788550325248308 1660044435849\nmacro,stock=AAPL Price=0.7887041778036982,RSI=0.9580521888642006,Volume=0.4591083111567962 1660044436849\nmacro,stock=NFLX Price=0.18822135657486783,RSI=0.07275710886951348,Volume=0.30987382389780815 1660044436849\nmacro,stock=MSFT Price=0.4520685755429874,RSI=0.35484084069748845,Volume=0.6088321900210175 1660044436849\nmacro,stock=GOOGL Price=0.06374901928702081,RSI=0.7859983366437384,Volume=0.30867527479308887 1660044436849\nmacro,stock=AAPL Price=0.6597067493291142,RSI=0.3253048056878177,Volume=0.07134675736922003 1660044437849\nmacro,stock=NFLX Price=0.9708557015197519,RSI=0.5542841051947816,Volume=0.8346282388391487 1660044437849\nmacro,stock=MSFT Price=0.6074777753931158,RSI=0.2624680251088799,Volume=0.5289297544414852 1660044437849\nmacro,stock=GOOGL Price=0.27192670144650966,RSI=0.40318891005577817,Volume=0.9236643221787052 1660044437849\nmacro,stock=AAPL Price=0.914487913056748,RSI=0.34708817096674927,Volume=0.355989595429149 1660044438849\nmacro,stock=NFLX Price=0.6754526865843875,RSI=0.7105176701991677,Volume=0.04647901249991515 1660044438849\nmacro,stock=MSFT Price=0.6235862051030008,RSI=0.40657665232765505,Volume=0.6394073752776313 1660044438849\nmacro,stock=GOOGL Price=0.8115581003524224,RSI=0.24062704237866317,Volume=0.1955898089247271 1660044438849\nmacro,stock=AAPL Price=0.41012086634172584,RSI=0.8362651719247415,Volume=0.6449882763195827 1660044439849\nmacro,stock=NFLX Price=0.04547242111821248,RSI=0.9513126085970982,Volume=0.18983535236993065 1660044439849\nmacro,stock=MSFT Price=0.7001841500138629,RSI=0.3609773268001528,Volume=0.4683295186641523 1660044439849\nmacro,stock=GOOGL Price=0.6400457568396799,RSI=0.6338321811676654,Volume=0.73847874268897 1660044439849\nmacro,stock=AAPL Price=0.867026884871651,RSI=0.9798937610322712,Volume=0.6984868701074483 1660044440849\nmacro,stock=NFLX Price=0.44782543030616384,RSI=0.1725301359260124,Volume=0.9949304996186534 1660044440849\nmacro,stock=MSFT Price=0.560186401820665,RSI=0.8070179565128155,Volume=0.9119658380635974 1660044440849\nmacro,stock=GOOGL Price=0.7634089714215698,RSI=0.23155010755601846,Volume=0.797300191254848 1660044440849\nmacro,stock=AAPL Price=0.4919580190846825,RSI=0.04833103486975521,Volume=0.5733175067756064 1660044441849\nmacro,stock=NFLX Price=0.7794539700246367,RSI=0.2875074203610679,Volume=0.11831508466091845 1660044441849\nmacro,stock=MSFT Price=0.38596115124106867,RSI=0.6856691881587962,Volume=0.889829497441647 1660044441849\nmacro,stock=GOOGL Price=0.20742893809218466,RSI=0.9962845374960013,Volume=0.38702770693246447 1660044441849\nmacro,stock=AAPL Price=0.954673451633858,RSI=0.23068092594788925,Volume=0.5134044793297395 1660044442849\nmacro,stock=NFLX Price=0.566033842971962,RSI=0.7635374400452328,Volume=0.9697896554109158 1660044442849\nmacro,stock=MSFT Price=0.2637313702152463,RSI=0.7504003145278523,Volume=0.9273412103576001 1660044442849\nmacro,stock=GOOGL Price=0.7586809472336218,RSI=0.0626334284995157,Volume=0.717091719097813 1660044442849\nmacro,stock=AAPL Price=0.5819144854437993,RSI=0.9111649897381449,Volume=0.17798773831972337 1660044443849\nmacro,stock=NFLX Price=0.9040519648275781,RSI=0.4045961483978202,Volume=0.36077605044448346 1660044443849\nmacro,stock=MSFT Price=0.04195882072818091,RSI=0.24335916979016503,Volume=0.8342672175681131 1660044443849\nmacro,stock=GOOGL Price=0.1014581279725747,RSI=0.0908023414510738,Volume=0.09213614081733257 1660044443849\nmacro,stock=AAPL Price=0.2775969021326844,RSI=0.4908117102723861,Volume=0.8368840186042374 1660044444849\nmacro,stock=NFLX Price=0.7953134957716285,RSI=0.9297516161201617,Volume=0.6966715360660173 1660044444850\nmacro,stock=MSFT Price=0.8498747785495513,RSI=0.19431231867875443,Volume=0.9101555537688967 1660044444850\nmacro,stock=GOOGL Price=0.4434923105280426,RSI=0.6327929412083354,Volume=0.2192172252682646 1660044444850\nmacro,stock=AAPL Price=0.35884829735528656,RSI=0.8178828309861498,Volume=0.0754399983627776 1660044445850\nmacro,stock=NFLX Price=0.07651878643787302,RSI=0.7852594815652344,Volume=0.220927308223977 1660044445850\nmacro,stock=MSFT Price=0.46182764821789146,RSI=0.3113067951872298,Volume=0.25348664676405275 1660044445850\nmacro,stock=GOOGL Price=0.40218672556282287,RSI=0.631893559669429,Volume=0.2246097764131747 1660044445850\nmacro,stock=AAPL Price=0.9834661735075051,RSI=0.9048796601837023,Volume=0.8920541882283451 1660044446850\nmacro,stock=NFLX Price=0.27915037310321367,RSI=0.7370977978506893,Volume=0.4336004512594883 1660044446850\nmacro,stock=MSFT Price=0.37489297966878155,RSI=0.5144448942076162,Volume=0.6229560404781662 1660044446850\nmacro,stock=GOOGL Price=0.18011970119763043,RSI=0.7184929799582587,Volume=0.5024162754872833 1660044446850\nmacro,stock=AAPL Price=0.250890479571954,RSI=0.24289826290713357,Volume=0.5137435409947008 1660044447850\nmacro,stock=NFLX Price=0.16824167870927742,RSI=0.6040412626016127,Volume=0.3149035609794162 1660044447850\nmacro,stock=MSFT Price=0.9107837292932687,RSI=0.8346015527833949,Volume=0.7691793466837485 1660044447850\nmacro,stock=GOOGL Price=0.49344328975970464,RSI=0.4655373949217596,Volume=0.7738913300822361 1660044447850\nmacro,stock=AAPL Price=0.43209852746965316,RSI=0.7180690451757343,Volume=0.18655086502410445 1660044448850\nmacro,stock=NFLX Price=0.8719949809897022,RSI=0.35209665522309486,Volume=0.8442121857481841 1660044448850\nmacro,stock=MSFT Price=0.6735731367942666,RSI=0.7488258628070313,Volume=0.05159717767199146 1660044448850\nmacro,stock=GOOGL Price=0.3845989313582354,RSI=0.7183490244934285,Volume=0.6018902133933317 1660044448850\nmacro,stock=AAPL Price=0.6428515502903155,RSI=0.3914058804812859,Volume=0.44607835090372083 1660044449850\nmacro,stock=NFLX Price=0.939761488788078,RSI=0.9477978791390171,Volume=0.4518165008341193 1660044449850\nmacro,stock=MSFT Price=0.041954310348878976,RSI=0.6701124255995752,Volume=0.1368004679127256 1660044449850\nmacro,stock=GOOGL Price=0.659196125337197,RSI=0.11868747468465057,Volume=0.8290850449513744 1660044449850\nmacro,stock=AAPL Price=0.1047468844145395,RSI=0.6332181619144168,Volume=0.16583320103700494 1660044450850\nmacro,stock=NFLX Price=0.46128770232302707,RSI=0.021577489090036872,Volume=0.8236269261494125 1660044450850\nmacro,stock=MSFT Price=0.34866687840852306,RSI=0.7012785591797683,Volume=0.5915560681691238 1660044450850\nmacro,stock=GOOGL Price=0.8592575394545932,RSI=0.06415717081818773,Volume=0.32408816446744837 1660044450850\nmacro,stock=AAPL Price=0.7779509158932891,RSI=0.4624234402781343,Volume=0.9363209304039787 1660044451850\nmacro,stock=NFLX Price=0.12997856395234098,RSI=0.21233229979456247,Volume=0.753457606686219 1660044451850\nmacro,stock=MSFT Price=0.790196658399053,RSI=0.572094910173328,Volume=0.8103121029346956 1660044451850\nmacro,stock=GOOGL Price=0.5999832961063822,RSI=0.24571941944073883,Volume=0.14128207198393206 1660044451850\nmacro,stock=AAPL Price=0.3290018804048439,RSI=0.5524140123887422,Volume=0.7499496422681763 1660044452850\nmacro,stock=NFLX Price=0.9435715244347836,RSI=0.20523955249484904,Volume=0.2783108048262761 1660044452850\nmacro,stock=MSFT Price=0.27869151695438443,RSI=0.572898984435737,Volume=0.42381517028862803 1660044452850\nmacro,stock=GOOGL Price=0.34131189967214937,RSI=0.013178456890358792,Volume=0.38190947936676867 1660044452850\nmacro,stock=AAPL Price=0.2798595322531383,RSI=0.31300710101992346,Volume=0.26610969835204024 1660044453850\nmacro,stock=NFLX Price=0.19988773823539208,RSI=0.5183384179568791,Volume=0.26211907486090835 1660044453850\nmacro,stock=MSFT Price=0.31464792084530857,RSI=0.14413242760957645,Volume=0.3662355107294366 1660044453850\nmacro,stock=GOOGL Price=0.28176869088907563,RSI=0.4659155838413047,Volume=0.14446922353962444 1660044453850\nmacro,stock=AAPL Price=0.8021369210686468,RSI=0.815685681486357,Volume=0.9600523911757042 1660044454850\nmacro,stock=NFLX Price=0.41264796986470165,RSI=0.8508073433951506,Volume=0.16821644582051398 1660044454850\nmacro,stock=MSFT Price=0.6058010893302271,RSI=0.21768302855260402,Volume=0.6344086457589324 1660044454850\nmacro,stock=GOOGL Price=0.8414171160784665,RSI=0.411384392832829,Volume=0.8607584402302999 1660044454850\nmacro,stock=AAPL Price=0.43972932708110335,RSI=0.12941919437484684,Volume=0.6484922092242599 1660044455850\nmacro,stock=NFLX Price=0.5133526590258135,RSI=0.9089765003124484,Volume=0.44519009286498357 1660044455850\nmacro,stock=MSFT Price=0.4977352615773366,RSI=0.7858532508268988,Volume=0.11653017855493897 1660044455850\nmacro,stock=GOOGL Price=0.08528276610807717,RSI=0.7246773905465312,Volume=0.7272455351769834 1660044455850\nmacro,stock=AAPL Price=0.8783282979892664,RSI=0.03909604087569285,Volume=0.9866996039159099 1660044456850\nmacro,stock=NFLX Price=0.7974343725049347,RSI=0.8138730253346574,Volume=0.3411786332456447 1660044456850\nmacro,stock=MSFT Price=0.6839442117844934,RSI=0.044203881050297156,Volume=0.5871730841118654 1660044456850\nmacro,stock=GOOGL Price=0.15267135700432777,RSI=0.4029996256174241,Volume=0.4491539839460027 1660044456850\nmacro,stock=AAPL Price=0.16789090791726669,RSI=0.7259156309808705,Volume=0.2514504152070798 1660044457850\nmacro,stock=NFLX Price=0.5470650524290966,RSI=0.7937229816271628,Volume=0.7624792517077172 1660044457850\nmacro,stock=MSFT Price=0.3810508322631786,RSI=0.0991825299946163,Volume=0.3512751435517808 1660044457850\nmacro,stock=GOOGL Price=0.9664444769181479,RSI=0.2622423272150114,Volume=0.783635663865986 1660044457850\nmacro,stock=AAPL Price=0.5179939959112511,RSI=0.500866271530876,Volume=0.31574614872562445 1660044458850\nmacro,stock=NFLX Price=0.8858564737296563,RSI=0.32443975067105524,Volume=0.39307833673861003 1660044458850\nmacro,stock=MSFT Price=0.6829023196396459,RSI=0.48720353036490127,Volume=0.5715747141034484 1660044458850\nmacro,stock=GOOGL Price=0.547762885202287,RSI=0.2521917114990023,Volume=0.8294175345443223 1660044458851\nmacro,stock=AAPL Price=0.40032275731121636,RSI=0.5027094401869211,Volume=0.8958799717099024 1660044459851\nmacro,stock=NFLX Price=0.29648070206423227,RSI=0.3689495747844439,Volume=0.3769792802393779 1660044459851\nmacro,stock=MSFT Price=0.1873505857185176,RSI=0.9223028773892986,Volume=0.4067582389272958 1660044459851\nmacro,stock=GOOGL Price=0.4660623861909877,RSI=0.33186805900842464,Volume=0.3657902480602323 1660044459851\nmacro,stock=AAPL Price=0.4926818304657027,RSI=0.44000143496147504,Volume=0.7693472924565793 1660044460851\nmacro,stock=NFLX Price=0.1847181245784698,RSI=0.683318329256625,Volume=0.685041168850202 1660044460851\nmacro,stock=MSFT Price=0.5163605628582889,RSI=0.11535378404370944,Volume=0.2929086823440493 1660044460851\nmacro,stock=GOOGL Price=0.26821720237570146,RSI=0.5685865502909342,Volume=0.16460813412267106 1660044460851\nmacro,stock=AAPL Price=0.7038676958660575,RSI=0.5367666627376334,Volume=0.6267056087739812 1660044461851\nmacro,stock=NFLX Price=0.9500151052535822,RSI=0.12911633531411282,Volume=0.6440074267082662 1660044461851\nmacro,stock=MSFT Price=0.4599691442833411,RSI=0.3238028480320052,Volume=0.08307261408938782 1660044461851\nmacro,stock=GOOGL Price=0.551159022063881,RSI=0.7853017298866739,Volume=0.4074727810181721 1660044461851\nmacro,stock=AAPL Price=0.1924256937488571,RSI=0.21282630287777338,Volume=0.7509760247135715 1660044462851\nmacro,stock=NFLX Price=0.8812087870649798,RSI=0.5895395108089733,Volume=0.386562989211481 1660044462851\nmacro,stock=MSFT Price=0.6415222451849654,RSI=0.8706590749193014,Volume=0.739399431443435 1660044462851\nmacro,stock=GOOGL Price=0.13108074250346513,RSI=0.5868015546855415,Volume=0.06339605954702843 1660044462851\nmacro,stock=AAPL Price=0.7174980455538135,RSI=0.14096094002722304,Volume=0.6445312698492239 1660044463851\nmacro,stock=NFLX Price=0.6793892130996698,RSI=0.1838235881779875,Volume=0.11560504340748379 1660044463851\nmacro,stock=MSFT Price=0.7491472762439192,RSI=0.4228125019804956,Volume=0.33553785309330864 1660044463851\nmacro,stock=GOOGL Price=0.32643909782480274,RSI=0.8849003094546091,Volume=0.21428350622554415 1660044463851\nmacro,stock=AAPL Price=0.1871137139415343,RSI=0.5730411026125042,Volume=0.8189664354414333 1660044464851\nmacro,stock=NFLX Price=0.5625464610871332,RSI=0.43169664915037675,Volume=0.9626986228515645 1660044464851\nmacro,stock=MSFT Price=0.6603799767546865,RSI=0.650528042672595,Volume=0.5944351980852944 1660044464851\nmacro,stock=GOOGL Price=0.16405288886245672,RSI=0.855455008557749,Volume=0.7292791375206742 1660044464851\nmacro,stock=AAPL Price=0.25292072609804317,RSI=0.6615832157630233,Volume=0.1499294000386865 1660044465851\nmacro,stock=NFLX Price=0.6865062159421003,RSI=0.346451848127297,Volume=0.5272328125553872 1660044465851\nmacro,stock=MSFT Price=0.9568767147673669,RSI=0.7057494382053032,Volume=0.6209905406723456 1660044465851\nmacro,stock=GOOGL Price=0.021674227943317192,RSI=0.8576192727776272,Volume=0.0825456012440906 1660044465851\nmacro,stock=AAPL Price=0.6904164413064887,RSI=0.7721744084887314,Volume=0.5200046672545486 1660044466851\nmacro,stock=NFLX Price=0.1322209715372078,RSI=0.5640242557252952,Volume=0.5762776723866778 1660044466851\nmacro,stock=MSFT Price=0.8205426269680668,RSI=0.9734322399438549,Volume=0.12642131177285287 1660044466851\nmacro,stock=GOOGL Price=0.06902661410843158,RSI=0.19428940121393845,Volume=0.09758425533289594 1660044466851\nmacro,stock=AAPL Price=0.3841481322951714,RSI=0.5652977824837386,Volume=0.02631337159445568 1660044467851\nmacro,stock=NFLX Price=0.02508583894366745,RSI=0.2881333460819816,Volume=0.6148106704838197 1660044467851\nmacro,stock=MSFT Price=0.598263284078705,RSI=0.15971919945437152,Volume=0.6858105278377816 1660044467851\nmacro,stock=GOOGL Price=0.8227287759596581,RSI=0.9300103753430008,Volume=0.218839304411249 1660044467851\nmacro,stock=AAPL Price=0.7042381469714211,RSI=0.9615091745995356,Volume=0.6250769024430923 1660044468851\nmacro,stock=NFLX Price=0.8451657866928173,RSI=0.5640387365675257,Volume=0.28224088983929774 1660044468851\nmacro,stock=MSFT Price=0.19896560771647898,RSI=0.9939133221846818,Volume=0.8540499442284961 1660044468851\nmacro,stock=GOOGL Price=0.6662736511871956,RSI=0.99967479200449,Volume=0.49276727586219804 1660044468851\nmacro,stock=AAPL Price=0.8961917347741809,RSI=0.8800802521188841,Volume=0.8591579716376679 1660044469851\nmacro,stock=NFLX Price=0.7341122692819865,RSI=0.6636867281304538,Volume=0.31246553558114354 1660044469851\nmacro,stock=MSFT Price=0.15730355411513064,RSI=0.44401801123957507,Volume=0.509097672693839 1660044469851\nmacro,stock=GOOGL Price=0.6802061566053158,RSI=0.07108244299029798,Volume=0.5532959310489554 1660044469851\nmacro,stock=AAPL Price=0.6400157649091797,RSI=0.008299074789551941,Volume=0.28451716426108886 1660044470851\nmacro,stock=NFLX Price=0.4446608934829901,RSI=0.08347692391694461,Volume=0.3311160700868365 1660044470851\nmacro,stock=MSFT Price=0.9841581721164102,RSI=0.4167959170749148,Volume=0.9745675534607161 1660044470851\nmacro,stock=GOOGL Price=0.7162510815206842,RSI=0.1049076068476672,Volume=0.7926761922569671 1660044470851\nmacro,stock=AAPL Price=0.2989712547875699,RSI=0.8521936690086188,Volume=0.929784477617663 1660044471851\nmacro,stock=NFLX Price=0.5943248037838437,RSI=0.4003877838755041,Volume=0.14667904367721218 1660044471851\nmacro,stock=MSFT Price=0.7661290283639899,RSI=0.16154485415735165,Volume=0.05726449819896928 1660044471851\nmacro,stock=GOOGL Price=0.11748452376242868,RSI=0.29694486736493453,Volume=0.4739203050221371 1660044471851\nmacro,stock=AAPL Price=0.6614099310518692,RSI=0.1625032037949291,Volume=0.1682880587739769 1660044472851\nmacro,stock=NFLX Price=0.515172294868325,RSI=0.573234584982883,Volume=0.2078528936184315 1660044472851\nmacro,stock=MSFT Price=0.23069572797436633,RSI=0.2900112122886933,Volume=0.802355458162643 1660044472851\nmacro,stock=GOOGL Price=0.2537115827320491,RSI=0.4704622325428287,Volume=0.7056394655584598 1660044472851\nmacro,stock=AAPL Price=0.6089242456200357,RSI=0.3337174664407838,Volume=0.4858902760919993 1660044473852\nmacro,stock=NFLX Price=0.1637195608925427,RSI=0.06785356846247481,Volume=0.7072275129977462 1660044473852\nmacro,stock=MSFT Price=0.3606940389781951,RSI=0.9040001460644705,Volume=0.519450799157602 1660044473852\nmacro,stock=GOOGL Price=0.4159502453526106,RSI=0.2892938320639121,Volume=0.6144252612792842 1660044473852\nmacro,stock=AAPL Price=0.12719068892507657,RSI=0.044264068464451634,Volume=0.6047096987320454 1660044474852\nmacro,stock=NFLX Price=0.27365533115505625,RSI=0.09030355307283222,Volume=0.9813702709260395 1660044474852\nmacro,stock=MSFT Price=0.9408499229704266,RSI=0.2509185935893976,Volume=0.5438554410116039 1660044474852\nmacro,stock=GOOGL Price=0.3607641387449204,RSI=0.4389371908550881,Volume=0.7064407283233353 1660044474852\nmacro,stock=AAPL Price=0.2993510299626002,RSI=0.9136757547416654,Volume=0.18952902445690745 1660044475852\nmacro,stock=NFLX Price=0.9374380400576214,RSI=0.44182380246661257,Volume=0.3572099960424381 1660044475852\nmacro,stock=MSFT Price=0.6656884420094104,RSI=0.7841606313549964,Volume=0.6723319757868695 1660044475852\nmacro,stock=GOOGL Price=0.8512989854740463,RSI=0.6075472686121518,Volume=0.00129891359927381 1660044475852\nmacro,stock=AAPL Price=0.6954620862917018,RSI=0.5197907718302232,Volume=0.580302765215279 1660044476852\nmacro,stock=NFLX Price=0.6785224353888978,RSI=0.27128215250750665,Volume=0.2626222010924828 1660044476852\nmacro,stock=MSFT Price=0.07334780218994985,RSI=0.5195651227740413,Volume=0.34184205527712863 1660044476852\nmacro,stock=GOOGL Price=0.2723167243166017,RSI=0.4781562142848932,Volume=0.8063751515800837 1660044476852\nmacro,stock=AAPL Price=0.3777501010340003,RSI=0.05582096969483319,Volume=0.14339773777519849 1660044477852\nmacro,stock=NFLX Price=0.5198534587204042,RSI=0.293225532143659,Volume=0.956423996626413 1660044477852\nmacro,stock=MSFT Price=0.233360797121537,RSI=0.032835913791792515,Volume=0.9936469897192588 1660044477852\nmacro,stock=GOOGL Price=0.7229700701786377,RSI=0.03039759331342129,Volume=0.28564646126464255 1660044477852\nmacro,stock=AAPL Price=0.025781135673753508,RSI=0.6542079385600682,Volume=0.7589263308204067 1660044478852\nmacro,stock=NFLX Price=0.39457443874598996,RSI=0.1522672194198642,Volume=0.9093209706359713 1660044478852\nmacro,stock=MSFT Price=0.7947894688376178,RSI=0.4178233138415054,Volume=0.6605856570704296 1660044478852\nmacro,stock=GOOGL Price=0.33016540630369007,RSI=0.7785600692723651,Volume=0.6794986289875302 1660044478852\nmacro,stock=AAPL Price=0.9930977295092388,RSI=0.3229665271967973,Volume=0.5554702513634575 1660044479852\nmacro,stock=NFLX Price=0.6450846123358565,RSI=0.19442542350389602,Volume=0.40112987453817495 1660044479852\nmacro,stock=MSFT Price=0.12563214684866908,RSI=0.7680774130829611,Volume=0.5219152576649441 1660044479852\nmacro,stock=GOOGL Price=0.661475317562441,RSI=0.8052617736143337,Volume=0.6558432766251943 1660044479852\nmacro,stock=AAPL Price=0.8273856684819697,RSI=0.6973416424552201,Volume=0.41396668418363847 1660044480852\nmacro,stock=NFLX Price=0.8445421608319421,RSI=0.09783400924684016,Volume=0.9875953015620718 1660044480852\nmacro,stock=MSFT Price=0.8865568818089427,RSI=0.7466976903402132,Volume=0.7779967738766499 1660044480852\nmacro,stock=GOOGL Price=0.7316568397812919,RSI=0.940125046132231,Volume=0.9546325463288897 1660044480852\nmacro,stock=AAPL Price=0.7295403413620571,RSI=0.03702724110521782,Volume=0.5427655172101864 1660044481852\nmacro,stock=NFLX Price=0.8004651197346454,RSI=0.06155038420662173,Volume=0.12194818727566148 1660044481852\nmacro,stock=MSFT Price=0.20394412818612506,RSI=0.07237553330537605,Volume=0.18255923290649112 1660044481852\nmacro,stock=GOOGL Price=0.28712495999581156,RSI=0.2540963227174413,Volume=0.5969051714487007 1660044481852\nmacro,stock=AAPL Price=0.6555407213083432,RSI=0.38211631712246696,Volume=0.7031767684736472 1660044482852\nmacro,stock=NFLX Price=0.3555813886884235,RSI=0.4670189831165521,Volume=0.7199686923278805 1660044482852\nmacro,stock=MSFT Price=0.6504566162035696,RSI=0.761827023230187,Volume=0.863256560690819 1660044482852\nmacro,stock=GOOGL Price=0.6378949691615318,RSI=0.8238876058295818,Volume=0.6226819473517525 1660044482852\nmacro,stock=AAPL Price=0.8163888829642802,RSI=0.2439554367599801,Volume=0.13977140752685957 1660044483852\nmacro,stock=NFLX Price=0.7334767049552181,RSI=0.07535903051552495,Volume=0.10326455455130912 1660044483852\nmacro,stock=MSFT Price=0.4729349074656558,RSI=0.7547680229405576,Volume=0.35335144543525376 1660044483852\nmacro,stock=GOOGL Price=0.11326553211447321,RSI=0.0295602052299071,Volume=0.3810748701200717 1660044483852\nmacro,stock=AAPL Price=0.3751071835564507,RSI=0.04351865817661171,Volume=0.5092423116138748 1660044484852\nmacro,stock=NFLX Price=0.29603886459421536,RSI=0.4014252141929323,Volume=0.11872634369780688 1660044484852\nmacro,stock=MSFT Price=0.9360214635206241,RSI=0.45935747627387613,Volume=0.12436769914501933 1660044484852\nmacro,stock=GOOGL Price=0.3602759457477728,RSI=0.6049680921076305,Volume=0.6872808633877762 1660044484852\nmacro,stock=AAPL Price=0.5708440128960447,RSI=0.3508548094708335,Volume=0.6185513315774481 1660044485853\nmacro,stock=NFLX Price=0.8979927274058247,RSI=0.9842313796288228,Volume=0.23508021885605224 1660044485853\nmacro,stock=MSFT Price=0.3182781251427861,RSI=0.5651718004986063,Volume=0.7435073978769695 1660044485853\nmacro,stock=GOOGL Price=0.9541364638254818,RSI=0.18395230801185913,Volume=0.7370957449562292 1660044485853\nmacro,stock=AAPL Price=0.959904781338298,RSI=0.05621532386957995,Volume=0.409075647480722 1660044486853\nmacro,stock=NFLX Price=0.025984833923871298,RSI=0.8764364893956395,Volume=0.1010549666465812 1660044486853\nmacro,stock=MSFT Price=0.5467539205343807,RSI=0.5492972275267114,Volume=0.6120426463520828 1660044486853\nmacro,stock=GOOGL Price=0.2697613060746653,RSI=0.4375450719680295,Volume=0.5516001716578294 1660044486853\nmacro,stock=AAPL Price=0.18950444614458328,RSI=0.8636686109851228,Volume=0.3600711662065905 1660044487853\nmacro,stock=NFLX Price=0.46066201582473076,RSI=0.8995800406315644,Volume=0.6495400004221185 1660044487853\nmacro,stock=MSFT Price=0.281582011881484,RSI=0.6767333702772111,Volume=0.7610284850175862 1660044487853\nmacro,stock=GOOGL Price=0.5039657231398456,RSI=0.8537856952300871,Volume=0.6995844940070154 1660044487853\nmacro,stock=AAPL Price=0.9656805826698193,RSI=0.6623922848138065,Volume=0.6540910462395014 1660044488853\nmacro,stock=NFLX Price=0.35592071837613815,RSI=0.7469994628866986,Volume=0.9519395897723761 1660044488853\nmacro,stock=MSFT Price=0.9055891162769355,RSI=0.3910923657663803,Volume=0.8179923238641404 1660044488853\nmacro,stock=GOOGL Price=0.6181884226128225,RSI=0.286515766015066,Volume=0.9956860617264521 1660044488853\nmacro,stock=AAPL Price=0.8729977252459554,RSI=0.4257494572476238,Volume=0.9990469184043835 1660044489853\nmacro,stock=NFLX Price=0.24897118628006765,RSI=0.5545528242638977,Volume=0.5672820720895588 1660044489853\nmacro,stock=MSFT Price=0.9169715956658914,RSI=0.24535267075388634,Volume=0.1500358091935895 1660044489853\nmacro,stock=GOOGL Price=0.8408744224129606,RSI=0.6369472781695922,Volume=0.44949249985537665 1660044489853\nmacro,stock=AAPL Price=0.8295455884210859,RSI=0.030320787164087704,Volume=0.4672560695157887 1660044490853\nmacro,stock=NFLX Price=0.2844443160384983,RSI=0.4547112745138844,Volume=0.7241019288343468 1660044490853\nmacro,stock=MSFT Price=0.35529162836493,RSI=0.4547187756459391,Volume=0.30868961691899655 1660044490853\nmacro,stock=GOOGL Price=0.8737210736921651,RSI=0.8452608546296432,Volume=0.8223844638370199 1660044490853\nmacro,stock=AAPL Price=0.594767490647442,RSI=0.8090795417608955,Volume=0.6621042526576408 1660044491853\nmacro,stock=NFLX Price=0.6328415267296107,RSI=0.7919130334496893,Volume=0.10002068976091627 1660044491853\nmacro,stock=MSFT Price=0.3951454807704857,RSI=0.7195374981492535,Volume=0.7120681504915071 1660044491853\nmacro,stock=GOOGL Price=0.06588891331082103,RSI=0.08874925052506255,Volume=0.1714107871634437 1660044491853\nmacro,stock=AAPL Price=0.08649616736083499,RSI=0.7691928697710034,Volume=0.7812521619312328 1660044492853\nmacro,stock=NFLX Price=0.05127855212791477,RSI=0.7942750583021382,Volume=0.6828850524854415 1660044492853\nmacro,stock=MSFT Price=0.5284389638043046,RSI=0.867460638071724,Volume=0.3670248466804287 1660044492853\nmacro,stock=GOOGL Price=0.1709614854847603,RSI=0.5116895422662039,Volume=0.3292748725577088 1660044492853\nmacro,stock=AAPL Price=0.4704059851661806,RSI=0.11712042123956057,Volume=0.5502822831958104 1660044493853\nmacro,stock=NFLX Price=0.7502156792468493,RSI=0.7494372431451923,Volume=0.8422820431725878 1660044493853\nmacro,stock=MSFT Price=0.5633762526988267,RSI=0.240867831951751,Volume=0.7809654125369621 1660044493853\nmacro,stock=GOOGL Price=0.6993465040975565,RSI=0.2278446059257997,Volume=0.18003956768383989 1660044493853\nmacro,stock=AAPL Price=0.7981687653044595,RSI=0.7624219216755305,Volume=0.5255982909628771 1660044494853\nmacro,stock=NFLX Price=0.9790213930460601,RSI=0.6966160252727086,Volume=0.2906084445394336 1660044494853\nmacro,stock=MSFT Price=0.3237236912877005,RSI=0.2828558937647524,Volume=0.015811732113927612 1660044494853\nmacro,stock=GOOGL Price=0.5599940310095158,RSI=0.590890312996021,Volume=0.7801070051136706 1660044494853\nmacro,stock=AAPL Price=0.017576360172883332,RSI=0.146479492991612,Volume=0.931630292076802 1660044495853\nmacro,stock=NFLX Price=0.777792440942867,RSI=0.5054237900747816,Volume=0.6664222437478168 1660044495853\nmacro,stock=MSFT Price=0.8776773875535253,RSI=0.6951557416256143,Volume=0.8525485493513841 1660044495853\nmacro,stock=GOOGL Price=0.01019959155108785,RSI=0.8164006326135472,Volume=0.4337248712239773 1660044495853\nmacro,stock=AAPL Price=0.9455692905298262,RSI=0.6537116899132743,Volume=0.8952026820269108 1660044496853\nmacro,stock=NFLX Price=0.21308640643523646,RSI=0.5341743518500449,Volume=0.5421286841527634 1660044496853\nmacro,stock=MSFT Price=0.4181701529032541,RSI=0.9811401058596033,Volume=0.2101240890904812 1660044496853\nmacro,stock=GOOGL Price=0.6386863211841135,RSI=0.3407761214346814,Volume=0.8439513308740014 1660044496853\nmacro,stock=AAPL Price=0.440479804024004,RSI=0.6929222884081596,Volume=0.5237467464333382 1660044497853\nmacro,stock=NFLX Price=0.054161231204324145,RSI=0.9622437021236033,Volume=0.9013250979023726 1660044497853\nmacro,stock=MSFT Price=0.832203307468771,RSI=0.5947870157378862,Volume=0.0739229544001403 1660044497853\nmacro,stock=GOOGL Price=0.14633106115600192,RSI=0.06541961047821743,Volume=0.9346013052512201 1660044497853\nmacro,stock=AAPL Price=0.1288696069808729,RSI=0.1548751288777792,Volume=0.1273597961155689 1660044498853\nmacro,stock=NFLX Price=0.11895333429581179,RSI=0.879024505614153,Volume=0.790239399780486 1660044498853\nmacro,stock=MSFT Price=0.559749299671082,RSI=0.6617316398708192,Volume=0.2706216093926028 1660044498853\nmacro,stock=GOOGL Price=0.6890579385432112,RSI=0.23819966456494313,Volume=0.2568558613652431 1660044498853\nmacro,stock=AAPL Price=0.5195450635326937,RSI=0.7548072872550067,Volume=0.2641514229771479 1660044499853\nmacro,stock=NFLX Price=0.37664208148056844,RSI=0.201348709446598,Volume=0.9206209171529462 1660044499853\nmacro,stock=MSFT Price=0.8154556134108817,RSI=0.1271307071042469,Volume=0.27914390806943856 1660044499853\nmacro,stock=GOOGL Price=0.4566665834734177,RSI=0.29372812599171483,Volume=0.9816057936875547 1660044499853\nmacro,stock=AAPL Price=0.5621005573931542,RSI=0.35695173363554655,Volume=0.8256749484622378 1660044500854\nmacro,stock=NFLX Price=0.3336738877952845,RSI=0.5289055169289032,Volume=0.2852983761335667 1660044500854\nmacro,stock=MSFT Price=0.9276090245131781,RSI=0.22635407550939224,Volume=0.6134231552571598 1660044500854\nmacro,stock=GOOGL Price=0.3111221360984673,RSI=0.8380392050394021,Volume=0.17981379040053713 1660044500854\nmacro,stock=AAPL Price=0.6714584538263026,RSI=0.5171785137706275,Volume=0.6032932439316494 1660044501854\nmacro,stock=NFLX Price=0.963459535478376,RSI=0.3441711279635673,Volume=0.626432097667863 1660044501854\nmacro,stock=MSFT Price=0.8111419141867244,RSI=0.9471527310107601,Volume=0.5982605568724108 1660044501854\nmacro,stock=GOOGL Price=0.3561151600992154,RSI=0.3929293500178275,Volume=0.8291476172256623 1660044501854\nmacro,stock=AAPL Price=0.8773566023641821,RSI=0.9053780401118382,Volume=0.18929917914115857 1660044502854\nmacro,stock=NFLX Price=0.058126118454618014,RSI=0.870008632022682,Volume=0.6609514541310642 1660044502854\nmacro,stock=MSFT Price=0.7105943469556301,RSI=0.4295421719299136,Volume=0.5239263137884712 1660044502854\nmacro,stock=GOOGL Price=0.2026798490954732,RSI=0.584623122889981,Volume=0.3029461327341628 1660044502854\nmacro,stock=AAPL Price=0.5122764438531046,RSI=0.8064668041766665,Volume=0.25494880961892297 1660044503854\nmacro,stock=NFLX Price=0.46415621860297207,RSI=0.02807166422382623,Volume=0.5658403870486973 1660044503854\nmacro,stock=MSFT Price=0.395224896154571,RSI=0.17666225299988003,Volume=0.10568685581935222 1660044503854\nmacro,stock=GOOGL Price=0.9362397875799335,RSI=0.08950247672957135,Volume=0.1631777010812988 1660044503854\nmacro,stock=AAPL Price=0.7181676584030314,RSI=0.8391259615693747,Volume=0.42603051436979555 1660044504854\nmacro,stock=NFLX Price=0.30235116887385427,RSI=0.9378562391738918,Volume=0.09476116123487088 1660044504854\nmacro,stock=MSFT Price=0.3250759823536715,RSI=0.48241319903546265,Volume=0.7628695762924907 1660044504854\nmacro,stock=GOOGL Price=0.04304693376108515,RSI=0.6232175656385239,Volume=0.26921206547078913 1660044504854\nmacro,stock=AAPL Price=0.7638311498008875,RSI=0.1841020406615479,Volume=0.8633928371096092 1660044505854\nmacro,stock=NFLX Price=0.7206050019403694,RSI=0.8566924132601369,Volume=0.06812530876642542 1660044505854\nmacro,stock=MSFT Price=0.20057460721716303,RSI=0.5995833791077742,Volume=0.1889113618701015 1660044505854\nmacro,stock=GOOGL Price=0.8201419682770646,RSI=0.5895209385679611,Volume=0.7379730879920282 1660044505854\nmacro,stock=AAPL Price=0.6501177743852435,RSI=0.2572006671676921,Volume=0.29276994081611685 1660044506854\nmacro,stock=NFLX Price=0.6037002982286178,RSI=0.6604805728793699,Volume=0.06840388225396277 1660044506854\nmacro,stock=MSFT Price=0.5385347828820125,RSI=0.7816808681764801,Volume=0.11987062969761897 1660044506854\nmacro,stock=GOOGL Price=0.2897740115918286,RSI=0.728008879024917,Volume=0.23890100871750986 1660044506854\nmacro,stock=AAPL Price=0.3916974300696262,RSI=0.7338858867099002,Volume=0.34317045860707573 1660044507854\nmacro,stock=NFLX Price=0.25501047166654,RSI=0.7560844663059867,Volume=0.9496377003547549 1660044507854\nmacro,stock=MSFT Price=0.18637822764474665,RSI=0.10220638182254349,Volume=0.11725380860685486 1660044507854\nmacro,stock=GOOGL Price=0.5254893364307675,RSI=0.8678107582430692,Volume=0.2758530904172547 1660044507854\nmacro,stock=AAPL Price=0.8631795329099959,RSI=0.626667649549492,Volume=0.18756080487692717 1660044508854\nmacro,stock=NFLX Price=0.07852234589684381,RSI=0.5504321745223788,Volume=0.2521421663323622 1660044508854\nmacro,stock=MSFT Price=0.5975898240658865,RSI=0.3745838516845654,Volume=0.9426493226610002 1660044508854\nmacro,stock=GOOGL Price=0.8455685548996409,RSI=0.2740375576651025,Volume=0.0392633567691516 1660044508854\nmacro,stock=AAPL Price=0.828601159189862,RSI=0.07624133688442136,Volume=0.8005115731718544 1660044509854\nmacro,stock=NFLX Price=0.6548896899866062,RSI=0.6527174149356667,Volume=0.4444041313675633 1660044509854\nmacro,stock=MSFT Price=0.21367403886501457,RSI=0.006507482449848556,Volume=0.549144835414972 1660044509854\nmacro,stock=GOOGL Price=0.6223770855505028,RSI=0.4451751865171133,Volume=0.6065071658440425 1660044509854\nmacro,stock=AAPL Price=0.50903042172937,RSI=0.6254193497250852,Volume=0.22799096342341885 1660044510854\nmacro,stock=NFLX Price=0.9580021567527529,RSI=0.12975191984236167,Volume=0.9159027403817914 1660044510854\nmacro,stock=MSFT Price=0.9955096395411843,RSI=0.5056949344253997,Volume=0.4234830440624453 1660044510854\nmacro,stock=GOOGL Price=0.7722212358900106,RSI=0.3871375202255618,Volume=0.41717046982454287 1660044510854\nmacro,stock=AAPL Price=0.4616054616756444,RSI=0.4855548623111222,Volume=0.564634704179936 1660044511854\nmacro,stock=NFLX Price=0.8684039482734266,RSI=0.6056912468308246,Volume=0.2357397035353459 1660044511854\nmacro,stock=MSFT Price=0.28962118003446513,RSI=0.4916731690211018,Volume=0.3773390666698323 1660044511854\nmacro,stock=GOOGL Price=0.9911024094504239,RSI=0.07910620421463477,Volume=0.6128596935064522 1660044511854'
Writing time is: 0.016311092000000027
disposed client.
```

Which version of the python do you use? 

Does the InfluxDB have enough resources?
- https://docs.influxdata.com/influxdb/v1.8/guides/hardware_sizing/#influxdb-oss-guidelines


Regards

**Comment by Aeroking21** [comment]
@bednar Sorry about that, it was a typo the client version I'm using is **1.31.0** and my Python version is **3.7.7** and the screenshot was an old version where I was using batches of **120**

This is the updated version:
![Screen Shot 2022-08-09 at 12 46 30](https://user-images.githubusercontent.com/55886814/183639467-e48413de-a819-4516-a549-71588f288c64.png)

I ran the code which you sent and it still takes over 2 minutes to write the full data, for example if I query the data using a simple flux query after 20 seconds of writing, it will still be uploading data and only after 2min will the whole data be uploaded. Is this expected? Maybe I'm misunderstanding the batching functionality?

**Comment by bednar** [comment]
The problem is in your timestamp. You are writing data in future: `simulated_time = datetime.datetime.utcnow() + datetime.timedelta(seconds=timestamp)`.

Try to use something like:

```python
now = datetime.datetime.utcnow() - datetime.timedelta(minutes=10)
for timestamp in range(120):
    for company_name in companies:
        for value in fields:
            temp = template
            temp["tags"]["stock"] = company_name
            temp["fields"][value] = random.random()

        # Generating simulated timestamps increasing with each outermost for loop
        simulated_time = now + datetime.timedelta(seconds=timestamp)
        temp["time"] = simulated_time

        dictionary_copy = copy.deepcopy(temp)
        point = Point.from_dict(dictionary_copy, WritePrecision.MS)  # Converting it to point
        list_points.append(point)
```

**Comment by Aeroking21** [comment]
That seems to work perfectly, I'm curious as to why one can't write data in the future ?

**Comment by bednar** [comment]
You can write data in future, but you have to also query data in future.

**Comment by Aeroking21** [comment]
Hi @bednar , I didn't want to open up another issue but when I scaled the timestamp to 10800 seconds, the data seems to upload perfectly. 

However when I'm querying it:
![Screen Shot 2022-08-10 at 00 05 31](https://user-images.githubusercontent.com/55886814/183776734-51e1d105-121b-4302-baed-7defc663586b.png)


I get a socket timeout error as it seems like this query takes >20seconds, I'm not sure how to increase the timeout functionality in the influx client:
![Screen Shot 2022-08-10 at 00 06 36](https://user-images.githubusercontent.com/55886814/183776970-16239260-c6e5-4a7b-a6a0-cea1b888ec93.png)

**Comment by Aeroking21** [comment]
Hi @bednar I didn't want to raise a new issue, but I was wondering whether the Async Write API allows batching of data?
